servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/GracefulConnectionClosureHandlingTest.java

```diff
@@ -479,6 +497,10 @@ private void assertClosedChannelException(ThrowingRunnable runnable, CloseEvent
         Exception e = assertThrows(ExecutionException.class, runnable);
         Throwable cause = e.getCause();
         assertThat(cause, instanceOf(ClosedChannelException.class));
+        if (protocol == HTTP_2) {
+            // HTTP/2 does not enhance ClosedChannelException with CloseEvent
```

Comment:
this is an unfortunate side effect of error handling being coupled to the close handler. we should consider a followup effort to preserve the exception.